src/openrct2/core/FileIndex.hpp

```diff
@@ -178,32 +180,82 @@ class FileIndex
         return ScanResult(stats, files);
     }
 
-    std::vector<TItem> Build(const ScanResult &scanResult) const
+    void BuildRange(const ScanResult &scanResult,
+                    size_t rangeStart,
+                    size_t rangeEnd,
+                    std::vector<TItem>& items,
+                    std::mutex& printLock) const
```

Comment:
Why not just return a new vector for the items rather than requiring the caller to supply it?